src/Appwrite/Platform/Workers/Migrations.php

```diff
@@ -301,16 +309,21 @@ protected function processMigration(Document $migration, Realtime $queueForRealt
                 $destination
             );
 
+            $aggregatedResources = [];
             /** Start Transfer */
             if (empty($source->getErrors())) {
                 $migration->setAttribute('stage', 'migrating');
                 $this->updateMigrationDocument($migration, $projectDocument, $queueForRealtime);
 
                 $transfer->run(
                     $migration->getAttribute('resources'),
-                    function () use ($migration, $transfer, $projectDocument, $queueForRealtime) {
+                    function ($resources) use ($migration, $transfer, $projectDocument, $queueForRealtime, &$aggregatedResources) {
                         $migration->setAttribute('resourceData', json_encode($transfer->getCache()));
                         $migration->setAttribute('statusCounters', json_encode($transfer->getStatusCounters()));
+
+                        if (!empty($resources)) {
+                            $aggregatedResources[] = $resources;
+                        }
```

Comment:
This could get very heavy on RAM if there are millions of docs, let's buffer and flush 1000 at a time. Doesn't actually matter if the migrations fails later - the docs that _were_ created are still in the DB